cluster-provision/images/vm-image-builder/build-containerdisk.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+SCRIPT_PATH=$(dirname "$(realpath "$0")")
+
+function cleanup() {
+  if [ $? -ne 0 ]; then
+    rm -f "${tar_file}"
+    docker image rm -f "${full_image_tag}"
+  fi
+
+  rm -rf "${temp_dir}"
+}
+
+IMAGE_NAME=$1
+TAG=$2
+VM_IMAGE_FILE=$3
+
+template_dockerfile="${SCRIPT_PATH}/Dockerfile.containerdisk"
+
+parent_dir=$(dirname "$VM_IMAGE_FILE")
+tar_file="${parent_dir}/${IMAGE_NAME}-${TAG}.tar"
+full_image_tag="${IMAGE_NAME}:${TAG}"
+
+trap 'cleanup' EXIT
+
+temp_dir=$(mktemp -d -p /tmp -t build.container-disk.XXXX)
```

Comment:
Who cleans the temp directory at the end?

Reply:
there is a trap but look on 
https://github.com/kubevirt/kubevirt/pull/4115#discussion_r484034330
bottom line we should return SIGTERM

Reply:
ohh I missed line 12.
SIGTERM is discusses in a different thread, no need to mention it here.